crates/llm-remote-http/src/lib.rs

```diff
@@ -0,0 +1,166 @@
+use anyhow::Result;
+use reqwest::{
+    header::{HeaderMap, HeaderValue},
+    Client, Url,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use spin_core::async_trait;
+use spin_llm::LlmEngine;
+use spin_world::llm::{self as wasi_llm, InferencingParams};
+use std::collections::HashSet;
+
+#[derive(Clone)]
+pub struct RemoteHttpLlmEngine {
+    auth_token: String,
+    url: Url,
+    client: Option<Client>,
+    allowed_models: HashSet<String>,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all(serialize = "camelCase"))]
+struct InferRequestBodyParams {
+    max_tokens: u32,
+    repeat_penalty: f32,
+    repeat_penalty_last_n_token_count: u32,
+    temperature: f32,
+    top_k: u32,
+    top_p: f32,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all(deserialize = "camelCase"))]
+struct InferUsage {
+    prompt_token_count: u32,
+    generated_token_count: u32,
+}
+#[derive(Deserialize)]
+struct InferResponseBody {
+    text: String,
+    usage: InferUsage,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all(deserialize = "camelCase"))]
+struct EmbeddingUsage {
+    prompt_token_count: u32,
+}
+
+#[derive(Deserialize)]
+struct EmbeddingResponseBody {
+    embeddings: Vec<Vec<f32>>,
+    usage: EmbeddingUsage,
+}
+
+#[async_trait]
+impl LlmEngine for RemoteHttpLlmEngine {
+    fn add_allowed_models(&mut self, models: HashSet<String>) {
```

Comment:
Seems like this is "set" rather than an "add" operation - if I call it twice it is not additive right?

Reply:
Agree - will rename.